src/lib/core/selection/pseudo-checkbox/pseudo-checkbox.ts

```diff
@@ -0,0 +1,53 @@
+import {
+  Component,
+  ViewEncapsulation,
+  Input,
+  ElementRef,
+  Renderer,
+} from '@angular/core';
+
+export type MdPseudoCheckboxState = 'unchecked' | 'checked' | 'indeterminate';
+
+/**
+ * Simplified checkbox without any of the underlying form control logic
+ * and fancy animations. To be used when composing other components.
+ * @docs-private
+ */
+@Component({
+  moduleId: module.id,
+  encapsulation: ViewEncapsulation.None,
+  selector: 'md-pseudo-checkbox',
+  styleUrls: ['pseudo-checkbox.css'],
+  template: '',
+  host: {
+    '[class.md-pseudo-checkbox-indeterminate]': 'state === "indeterminate"',
+    '[class.md-pseudo-checkbox-checked]': 'state === "checked"',
+    '[class.md-pseudo-checkbox-disabled]': 'disabled',
+  },
+})
+export class MdPseudoCheckbox {
+  /** Display state of the checkbox. */
+  @Input() state: MdPseudoCheckboxState = 'unchecked';
+
+  /** Whether the checkbox is disabled. */
+  @Input() disabled: boolean = false;
+
+  /** Color of the checkbox. */
+  @Input()
```

Comment:
Instead of doing an `@Input` for `color` and `disabled`, perhaps we should just include the class directly? My reasoning is that it more clearly communicates that it's not a real checkbox and is only for show. 

@kara thoughts?